Disco_Tie/code.py

```diff
@@ -0,0 +1,237 @@
+from adafruit_bluefruit_connect.color_packet import ColorPacket
+from adafruit_bluefruit_connect.button_packet import ButtonPacket
+
+uart_server = UARTServer()
+
+# User input vars
+mode = 0 # 0=audio, 1=rainbow, 2=larsen_scanner, 3=solid
+user_color= (127,0,0)
+speed = 6.0 # for larsen scanner
+
+# Audio meter vars
+PEAK_COLOR = (100, 0, 255)
+NUM_PIXELS = 10
+CURVE = 2
+SCALE_EXPONENT = math.pow(10, CURVE * -0.1)
+NUM_SAMPLES = 160
+
+# Restrict value to be between floor and ceiling.
+def constrain(value, floor, ceiling):
+    return max(floor, min(value, ceiling))
+
+# Scale input_value between output_min and output_max, exponentially.
+def log_scale(input_value, input_min, input_max, output_min, output_max):
+    normalized_input_value = (input_value - input_min) / \
+                             (input_max - input_min)
+    return output_min + \
+        math.pow(normalized_input_value, SCALE_EXPONENT) \
+        * (output_max - output_min)
+
+# Remove DC bias before computing RMS.
+def normalized_rms(values):
+    minbuf = int(mean(values))
+    samples_sum = sum(
+        float(sample - minbuf) * (sample - minbuf)
+        for sample in values
+    )
+
+    return math.sqrt(samples_sum / len(values))
+
+def mean(values):
+    return sum(values) / len(values)
+
+def volume_color(volume):
+    return 200, volume * (255 // NUM_PIXELS), 0
+
+# Set up NeoPixels and turn them all off.
+pixels = neopixel.NeoPixel(board.A1, NUM_PIXELS, brightness=0.1, auto_write=False)
+pixels.fill(0)
+pixels.show()
+
+mic = audiobusio.PDMIn(board.MICROPHONE_CLOCK, board.MICROPHONE_DATA,
+                       sample_rate=16000, bit_depth=16)
+
+# Record an initial sample to calibrate. Assume it's quiet when we start.
+samples = array.array('H', [0] * NUM_SAMPLES)
+mic.record(samples, len(samples))
+# Set lowest level to expect, plus a little.
+input_floor = normalized_rms(samples) + 10
+# Corresponds to sensitivity: lower means more pixels light up with lower sound
+input_ceiling = input_floor + 500
+peak = 0
+
+def wheel(wheel_pos):
+    # Input a value 0 to 255 to get a color value.
+    # The colours are a transition r - g - b - back to r.
+    if wheel_pos < 0 or wheel_pos > 255:
+        r = g = b = 0
+    elif wheel_pos < 85:
+        r = int(wheel_pos * 3)
+        g = int(255 - wheel_pos*3)
+        b = 0
+    elif wheel_pos < 170:
+        wheel_pos -= 85
+        r = int(255 - wheel_pos*3)
+        g = 0
+        b = int(wheel_pos*3)
+    else:
+        wheel_pos -= 170
+        r = 0
+        g = int(wheel_pos*3)
+        b = int(255 - wheel_pos*3)
+    return (r, g, b)
+
+def rainbow_cycle(wait):
+    for j in range(255):
+        for i in range(NUM_PIXELS):
+            pixel_index = (i * 256 // NUM_PIXELS) + j
+            pixels[i] = wheel(pixel_index & 255)
+        pixels.show()
+        time.sleep(wait)
+
+def audio_meter():
+    mic.record(samples, len(samples))
+    magnitude = normalized_rms(samples)
+    global peak
```

Comment:
You can pass peak in and return it rather than changing a global. That is more Pythonic.